lib/core/base/virtual-node.js

```diff
@@ -76,6 +76,15 @@ class VirtualNode {
 		return this.actualNode.hasAttribute(attrName);
 	}
 
+	/**
+	 * Get the input type. This is more complicated when you don't have an
+	 * actualNode to reference.
+	 * @return {String|undefined}
+	 */
+	get elementType() {
+		return this.actualNode.type;
```

Comment:
Just a bikeshed, but `|| null` would be preferred to undefined here.

Reply:
`null` is explicit, while `undefined` is not. IMO it's better to return a defined falsy value. It shows intention and follows practices put in place by the native DOM:

```js
document.createElement('div').getAttribute('type')
// => null
```

Reply:
But we're not accessing the `type` attribute, we're accessing the `type` property, which returns `undefined`.

```
document.createElement('div').type
// => undefined
```

Reply:
Good point.